server/src/internalClusterTest/java/org/elasticsearch/indices/settings/UpdateSettingsIT.java

```diff
@@ -687,4 +690,40 @@ private void runTestDefaultNumberOfReplicasTest(final boolean closeIndex) {
         assertThat(IndexMetadata.INDEX_NUMBER_OF_REPLICAS_SETTING.get(response.getIndexToSettings().get("test")), equalTo(1));
     }
 
+    public void testNoopUpdate() {
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        final ClusterService clusterService = internalCluster().getMasterNodeInstance(ClusterService.class);
+        assertAcked(client().admin().indices().prepareCreate("test")
+            .setSettings(Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 2)));
+        client().admin().cluster().prepareHealth()
+            .setWaitForNoInitializingShards(true)
+            .setWaitForEvents(Priority.LANGUID)
+            .setTimeout(TimeValue.MAX_VALUE)
+            .get();
+
+        ClusterState currentState = clusterService.state();
+        assertAcked(client().admin().indices().prepareUpdateSettings("test")
+            .setSettings(Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)));
+        assertNotSame(currentState, clusterService.state());
+        currentState = clusterService.state();
+
+        assertAcked(client().admin().indices().prepareUpdateSettings("test")
+            .setSettings(Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)));
+        assertSame(clusterService.state(), currentState);
+
+        assertAcked(client().admin().indices().prepareUpdateSettings("test")
+            .setSettings(Settings.builder().putNull(IndexMetadata.SETTING_NUMBER_OF_REPLICAS)));
+        assertSame(clusterService.state(), currentState);
+
+        assertAcked(client().admin().indices().prepareUpdateSettings("test")
+            .setSettings(Settings.builder()
+                .put(SETTING_BLOCKS_READ, true)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)));
+        assertNotSame(currentState, clusterService.state());
+        currentState = clusterService.state();
+
+        assertAcked(client().admin().indices().prepareUpdateSettings("test")
+            .setSettings(Settings.builder().put(SETTING_BLOCKS_READ, true)));
+        assertNotSame(currentState, clusterService.state());
```

Comment:
Is there a reason we do not want this to be the same? With the block already in place, this could be a no-op?

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/61348/commits/f43c2ead444482eebbd402773e847eb1a9e086b6.